lib/utils/types.ts

```diff
@@ -111,17 +113,20 @@ export function strictInterfaceWithOptionals<
 >(
   required: R,
   optional: O,
-  name: string
-): t.Type<t.mixed, t.InterfaceOf<R> & t.PartialOf<O>> {
+  name?: string
+): t.Type<
+  { [K in keyof R]: t.TypeOf<R[K]> } & { [K in keyof O]?: t.TypeOf<O[K]> },
+  { [K in keyof R]: t.OutputOf<R[K]> } & { [K in keyof O]?: t.OutputOf<O[K]> }
+> {
   const loose = t.intersection([t.interface(required), t.partial(optional)]);
   const props = Object.assign({}, required, optional);
   return new t.Type(
-    name,
-    (v): v is t.InterfaceOf<R> & t.PartialOf<O> =>
-      loose.is(v) &&
-      Object.getOwnPropertyNames(v).every(k => props.hasOwnProperty(k)),
-    (s, c) =>
-      loose.validate(s, c).chain(o => {
+    name || `StrictInterfaceWithOptionals(${loose.name})`,
```

Comment:
...just imagine the questions of somebody looking at this code for the first time :)